src/server/main_service.cc

```diff
@@ -65,7 +65,7 @@ struct MaxMemoryFlag {
 static bool AbslParseFlag(std::string_view in, MaxMemoryFlag* flag, std::string* err);
 static std::string AbslUnparseFlag(const MaxMemoryFlag& flag);
 
-ABSL_FLAG(uint32_t, port, 6379, "Redis port");
+ABSL_FLAG(int32_t, port, 6379, "Redis port");
```

Comment:
please extend the description of this flag, mention that 0 disables TCP port, -1 allocates it randomly from the available unbinded ports